sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample8_ConsumeEventsByBatch.cs

```diff
@@ -0,0 +1,135 @@
+                await using (EventHubProducer producer = client.CreateProducer(new EventHubProducerOptions { PartitionId = firstPartition }))
+                {
+                    // Because our consumer is reading from the latest position, it won't see events that have previously
+                    // been published. Before we can publish the events, we will need to ask the consumer to perform an operation,
+                    // because it opens its connection only when it needs to.  The first receive that we ask of it will not see
+                    // any events, but will allow the consumer to start watching the partition.
+
+                    await consumer.ReceiveAsync(1, TimeSpan.Zero);
+
+                    // Now that the consumer is watching the partition, let's publish a fair sized batch of events
+                    // we would like for it to consume.
+
+                    int eventBatchSize = 100;
+
+                    IEnumerable<EventData> eventBatch =
+                        Enumerable
+                          .Range(0, eventBatchSize)
+                          .Select(index => new EventData(Encoding.UTF8.GetBytes($"I am event #{ index }")));
+
+                    await producer.SendAsync(eventBatch);
+                    Console.WriteLine($"The event batch with { eventBatchSize } events has been published.");
+
+                    // To allow for throughput in our application, we will process the published events by consuming them in
+                    // small batches with a small wait time.  This will allow us to receive and process more quickly than blocking
+                    // to wait on a larger batch size.
+                    //
+                    // The values that are used in this example are inteded just for illustration and not as guidance for a recommended
+                    // set of defaults.  In a real-world applicaiton, determining the right balance of batch size and wait time to achieve
+                    // the desired performance will often vary depending on the application and event data being consumed.  It is an area
+                    // where some experimentation in the application context may prove helpful.
+                    //
+                    // Our example will attempt to read about 1/5 of the batch at a time, which should result in the need for 5 batches to
+                    // be consumed.  Because publishing and receving events is asynchronous, the events that we published may not be
+                    // immediately available for our consumer to see. To compensate, we will allow for a small number of extra attempts beyond
+                    // the expected 5 to to be sure that we don't stop reading before we receive all of our events.
+
+                    var receivedEvents = new List<EventData>();
```

Comment:
you might want to pass capacity to the list ctor.

Reply:
Were this production code, I'd agree.   I can probably count the number of times that I've seen the size passed to a list on construction in the wild on one hand, so in this case, I'm going to lean toward the less efficient familiar.